### DISCUSSION TITLE
output_ui rendered at startup and upon input change

### DISCUSSION THREAD

**Discussion by pisistrato** [posit-dev/py-shiny#1606] [Q&A]
I come from the R world and cannot figure out how to do this in Python.
 Look  at this dummy example
```
from shiny import App, Inputs, Outputs, Session, reactive, render, ui

app_ui = ui.page_fluid(
    ui.input_action_button("change", "Change"),
    ui.output_ui("Controls"),
)

def server(input: Inputs, output: Outputs, session: Session):
    @render.ui
    def Controls():
        return ui.TagList(
            ui.input_slider("Initial", "N", min=1, max=1000, value=500),
            ui.input_text("label", "Label"),
        )

    @render.ui
    @reactive.event(input.change)
    def Controls():
        return ui.TagList(
            ui.input_slider("Changed", "N", min=1, max=100, value=50),
            ui.input_text("label", "Label new"),
        )


app = App(app_ui, server)
```

How can I render `Controls` either at startup and re-render it when the action_button is clicked?

Just to clarify, I know one could update the slider and text, but I really want to render dynamically the ui

**Comment by QuintonJBaker** [comment]
Old thread at this point, but the fix is to put `ignore_init=true` as an option for the reactive event. That way the calculation won't fire the first time the event triggers, which is on initialization. 

Another tip since you (were 2 years ago) are new to Python. Functions and methods should generally start with a lower case letter. Classes start with capital letters.
